src/components/Stopwatch.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useRef, useState } from "react";
+import Delete from "../assets/delete-red.svg";
+import { convertTime } from "../helpers/timeConverter";
+
+interface StopwatchProps {
+  id: number;
+  onDelete: (idToDelete: number) => void;
+}
+
+const Stopwatch = ({ id, onDelete }: StopwatchProps) => {
+  const prevTime = localStorage.getItem(id.toString()) || "0";
+
+  const [started, setStarted] = useState(prevTime !== "0");
+  const [running, setRunning] = useState(false);
+  const [time, setTime] = useState(parseInt(prevTime));
+  const intervalId = useRef(0);
+
+  useEffect(() => {
+    setTimeout(() => {
+      localStorage.setItem(id.toString(), time.toString());
+    }, 250);
+  }, [id, time]);
+
+  function handleStart() {
+    setStarted(true);
+    setRunning(true);
+    intervalId.current = setInterval(() => setTime((prev) => prev + 1), 10);

```

Comment:
I believe, you need to use cleanup function on unmount (see previous comments), or you can get the case, when you delete a stopwatch, component is unmounted, but interval is not cleared